### ISSUE TITLE
Focused Launch: add error handling

### ISSUE THREAD

**Issue by razvanpapadopol** [Automattic/wp-calypso#47782]
**Blocked** until https://github.com/Automattic/wp-calypso/issues/48165 is closed.

___

#### Why
* The user should be notified as soon as something goes wrong.
* They shouldn't be stuck on a non-responsive screen.

#### What
* UI should always be updated after a failed API call.

#### Acceptance criteria
- [ ] Check if we handle errors for other API calls 
  - site data
  - plans
  - domain suggestions
  - getting / setting cart

**Comment by ciampo** [comment]
Should we close this issue, since it's already been tracked in 2 separate sub-issues?

**Comment by StefanNieuwenhuis** [comment]
> Should we close this issue, since it's already been tracked in 2 separate sub-issues?

@tjcafferkey We were discussing exactly this :) 
I think we can since this issue is adding a clogging layer of complexity to our project's board and doesn't add much.

**Comment by razvanpapadopol** [comment]
@StefanNieuwenhuis 
I've kept here just the parts that are not already handled somewhere else.

> * [ ]  Check if we handle errors for other API calls
>   
>   * site data
>   * plans
>   * domain suggestions
>   * getting / setting cart

You can remove this from #48166 to un-block that one and ship the Launch Site error handling sooner.

**Comment by tjcafferkey** [comment]
Yep in theory if a parent issues contains "sub tasks" that can be worked on independently then these should be tracked as separate issues and the "parent" issue should be closed as it then becomes redundant.

This prevents parent issues lingering in "In progress" for too long and accurately represents the work completed as we can close the independent (sub) issues as work gets completed.

Also if one sub issue is blocking another we can use the "blocked" label and reference it in the description "Blocked until [issue number] is merged"
